lib/Sema/TypeCheckAccess.cpp

```diff
@@ -681,6 +682,15 @@ void AccessControlChecker::check(Decl *D) {
     const TypeRepr *complainRepr = nullptr;
     auto downgradeToWarning = DowngradeToWarning::No;
 
+    // FIXME: Hack to ensure that we've computed the types involved here.
```

Comment:
Isn't this just the same as folding the request into the for_each below, instead of directly inspecting the TypeLoc?

Reply:
Yeah, but I need the index for the request. I’m looking for a better API design here, because TypeLoc is an anti-pattern in the request-evaluator world. I think we want an API to visit the inherited list and provide (Type, TypeRepr*) pairs at a given type resolution stage.